lib/snitcher/api/key.rb

```diff
@@ -0,0 +1,59 @@
+require "pp"
+
+require "net/https"
+require "timeout"
+require "base64"
+require "json"
+
+require "snitcher/api"
+require "snitcher/api/base"
+require "snitcher/version"
+
+class Snitcher::API::Key < Snitcher::API::Base
```

Comment:
For this I was thinking we would just have `Snitcher::API.get_key(username: "", password: "")` which would do a one off HTTP request and return the key. To do that we really don't need a full REST client to make that request. From there, key management (a feature we still need to build) would work using an existing key and the existing client.

I just feel that splitting the client into two to handle this one case is a bit overkill. It's okay to have some small duplication if it means a simpler implementation.